pkg/ingester/limiter.go

```diff
@@ -210,34 +211,46 @@ func (l *Limiter) convertGlobalToLocalLimit(userID string, globalLimit int) int
 		return 0
 	}
 
-	// Given we don't need a super accurate count (ie. when the ingesters
-	// topology changes) and we prefer to always be in favor of the tenant,
-	// we can use a per-ingester limit equal to:
-	// (global limit / number of ingesters) * replication factor
-	numIngesters := l.ring.HealthyInstancesCount()
-
-	// May happen because the number of ingesters is asynchronously updated.
-	// If happens, we just temporarily ignore the global limit.
-	if numIngesters == 0 {
-		return 0
+	zonesCount := l.getZonesCount()
+	var ingestersInZoneCount int
+	var ingestersInZoneCountShardFn func(int, int) int
```

Comment:
We don't need this function... `ShuffleShardExpectedInstancesPerZone` works fine for both cases.